packages/web-react/src/components/Collapse/Collapse.tsx

```diff
@@ -1,92 +1,47 @@
-    children,
-    isCollapsed,
-    hideOnCollapse,
-    renderTrigger,
-    ...rest
-  } = props;
-
-  const { collapsed, toggleHandler } = useCollapse({
-    isCollapsed,
-  });
-  const { wrapperClassName, contentClassName, triggerClassName } = useCollapseStyleProps({
-    isCollapsed: collapsed,
-  });
-  const { wrapperProps, contentProps, triggerProps } = useCollapseAriaProps({
-    id,
-    collapsibleToBreakpoint,
-    toggleHandler,
-    isCollapsed: collapsed,
-  });
-
-  const { styleProps: wrapperStyleProps, props: wrapperOtherProps } = useStyleProps({ ...rest });
-  const { styleProps: triggerStyleProps } = useStyleProps({
-    UNSAFE_className: triggerClassName,
-  });
-
-  const content = createElement(
-    contentElementType,
-    {
-      ...contentProps,
-      className: contentClassName,
-    },
-    children,
-  );
-
-  const triggerRenderHandler = () => {
-    const show = hideOnCollapse ? !(hideOnCollapse && collapsed) : true;
-    if (renderTrigger && show)
-      return renderTrigger({
-        collapsed,
-        trigger: {
-          ...triggerStyleProps,
-          ...triggerProps,
-        },
-      });
+const defaultProps = {
+  isCollapsed: false,
+  collapsibleToBreakpoint: undefined,
+};
 
-    return null;
-  };
+const Collapse = (props: SpiritCollapseProps) => {
+  const { children, ...restProps } = props;
 
-  const wrapperRenderHandler = () => {
-    if (hideOnCollapse && collapsed) {
-      return children;
-    }
+  const collapseElementRef: MutableRefObject<HTMLElement | undefined> = useRef();
+  const collapseHeight = useResizeHeight(collapseElementRef);
 
-    return createElement(
-      elementType || 'div',
-      {
-        ...wrapperOtherProps,
-        ...wrapperStyleProps,
-        ...wrapperProps,
-        className: classNames(wrapperClassName, wrapperStyleProps.className),
-        style: {
-          ...wrapperProps.style,
-          ...wrapperStyleProps.style,
-        },
-      },
-      content,
-    );
+  const { classProps } = useCollapseStyleProps(restProps.isCollapsed);
+  const { ariaProps, props: otherProps } = useCollapseAriaProps(restProps);
+  const { styleProps, props: transferProps } = useStyleProps(otherProps);
+  const collapseStyleProps = {
+    className: styleProps.className,
+    ...{ style: { height: restProps.isCollapsed ? collapseHeight : 0, ...styleProps.style } },
   };
 
   return (
-    <>
-      {triggerRenderHandler()}
-      {wrapperRenderHandler()}
-    </>
+    <div
+      {...transferProps}
+      {...collapseStyleProps}
+      {...ariaProps.root}
+      className={classNames(classProps.root, styleProps.className)}
+    >
+      {/* error TS2322: Type 'MutableRefObject<HTMLElement | undefined>' is not assignable to type 'LegacyRef<HTMLDivElement> | undefined'. */}
+      {/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
+      {/* @ts-ignore */}
+      <div ref={collapseElementRef} className={classProps.content}>
```

Comment:
Try also import `LegacyRef` and set `ref={collapseElementRef as LegacyRef<HTMLDivElement>}`.
Then you don't need those `@ts-ignore` lines anymore

React Ref has two types. One is what you set the constant with, and then the other is for the element on which you set this constant

Reply:
Thanks for your suggestion. After a short search, I think I found solution here: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/35572